packages-native/actions/src/camera/TakePictureAdvanced.ts

```diff
@@ -0,0 +1,225 @@
+        if (!picture.inheritsFrom("System.FileDocument")) {
+            const entity = picture.getEntity();
+            return reject(new Error(`Entity ${entity} does not inherit from 'System.FileDocument'`));
+        }
+        if (pictureQuality === "custom" && !maximumHeight && !maximumWidth) {
+            return reject(new Error("Picture quality is set to 'Custom', but no maximum width or height was provided"));
+        }
+        createMxObject("NativeMobileResources.TakePictureData").then((resultObject: any) => {
+            takePicture()
+                .then((response: any) => {
+                    if (!response || !response.uri) {
+                        return resolve(resultObject);
+                    }
+                    storeFile(picture, response.uri).then(pictureTaken => {
+                        resultObject.set("PictureTaken", pictureTaken);
+                        resultObject.set("URI", response.uri);
+                        resultObject.set("IsVertical", response.isVertical);
+                        resultObject.set("Width", response.width);
+                        resultObject.set("Height", response.height);
+                        resultObject.set("FileName", response.fileName);
+                        resultObject.set("FileSize", response.fileSize);
+                        resultObject.set("FileType", response.type);
+                        resolve(resultObject);
+                    });
+                })
+                .catch(error => {
+                    if (error === "canceled") {
+                        resolve(resultObject);
+                    } else {
+                        throw new Error(error);
+                    }
+                });
+        });
+    });
+
+    function takePicture(): Promise<string | undefined> {
+        return new Promise((resolve, reject) => {
+            const options = getOptions();
+            const method = getPictureMethod();
+
+            method(options, response => {
+                if (response.didCancel) {
+                    return resolve();
+                }
+
+                if (response.error) {
+                    const unhandledError = handleImagePickerError(response.error);
+                    if (!unhandledError) {
+                        return resolve();
+                    }
+                    return reject(new Error(response.error));
+                }
+
+                return resolve(response.uri);
+            });
+        });
+    }
+
+    function storeFile(imageObject: mendix.lib.MxObject, uri: string): Promise<boolean> {
+        return new Promise((resolve, reject) => {
+            fetch(uri)
+                .then(res => res.blob())
+                .then(blob => {
+                    const guid = imageObject.getGuid();
+                    // eslint-disable-next-line no-useless-escape
+                    const filename = /[^\/]*$/.exec(uri)![0];
+                    const onSuccess = (): void => resolve(true);
```

Comment:
Same here

Reply:
This bit I took from the original TakePicture. I want to keep the two the same where possible.